sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -1914,9 +1914,12 @@ object SQLConf {
     .createOptional
 
   val V2_SESSION_CATALOG = buildConf("spark.sql.catalog.session")
-      .doc("Name of the default v2 catalog, used when a catalog is not identified in queries")
+      .doc("A catalog implementation which will be used as the session catalog for Spark " +
+        "queries. The implementation must either extend `CatalogExtension`, or implement " +
+        "`CatalogPlugin`. When extending `CatalogExtension`, the default session catalog " +
```

Comment:
The class needs to implement `CatalogPlugin` to be loaded. I would rephrase this to "the implementation may extend `CatalogExtension` to be passed the default session catalog so that . . .".